python/lsst/sims/maf/slicers/baseSpatialSlicer.py

```diff
@@ -59,7 +60,10 @@ def __init__(self, verbose=True, lonCol='fieldRA', latCol='fieldDec',
         self.slicePoints['ra'] = None
         self.slicePoints['dec'] = None
         self.nslice = None
+        self.shape = None
         self.plotFuncs = [BaseHistogram, BaseSkyMap]
+        # Values in self.slicePoints to pass whole to the metrics
+        self.bulkSlicePointKeys = []
```

Comment:
I wonder if that will lead to a way to simplify complex metrics :)
How do the metrics give feedback to the slicer to change the plotFuncs
though? (I should go look at the code I suspect ..)

On Tue, Sep 29, 2015 at 3:16 PM Peter Yoachim <notifications@github.com>
wrote:

> In python/lsst/sims/maf/slicers/baseSpatialSlicer.py
> <https://github.com/lsst/sims_maf/pull/44#discussion_r40737795>:
>
> >          self.plotFuncs = [BaseHistogram, BaseSkyMap]
> > +        # Values in self.slicePoints to pass whole to the metrics
> > +        self.bulkSlicePointKeys = []
>
> So I started fiddling, and eventually went whole-hog and moved the bin
> information over to the metrics, and gave the metrics a shape attribute as
> well. This way, you can even run regular metrics and vector metrics with
> the same slicer!
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/lsst/sims_maf/pull/44/files#r40737795>.
>